### DISCUSSION TITLE
Deploying migrations and perform seeding on multi tenants database

### DISCUSSION THREAD

**Discussion by baijanathTharu** [prisma/prisma#17569] [Q&A] [answered]
We are using database per tenant in our multi tenant architecture. We can deploy the migrations from the pipeline and perform seeding as well.

However, when a new tenant has to be created, there are certain tenant specific data that has to be seeded. How can we pass that data to the `seed.ts` file when running the seed command?

**Comment by sabinadams** [comment]
What does the full development workflow look like in a scenario like this? There are potentially many ways to going about achieving this behavior but it really depends on how you are deploying the tenants.

**Reply by baijanathTharu** [reply]
In our application, we make Prisma instances for each tenant and we access the particular instance depending on the `tenantId` of the user. We have a separate application from which we create a new tenant. 
When we create a new one, we generate resources such as database, bucket etc for it.When the database is generated for the tenant, a pipeline is invoked from which migration and seeding has to be performed. We pull the repository containing the Prisma folder in the pipeline and run the `prisma migrate deploy` and `prisma db seed`  commands passing the database URL of the tenant as environment variable. The seed file contains only the static data that is common to all the tenants. However, there are some data that depends on the tenant such as `tenantName` that has to be seeded.

**Reply by sabinadams** [reply]
Would it be possible during that pipeline to pull in a static seed script file from S3 (or similar) specific to that tenant?

Generate DB Resource -> Pull Repository -> Pull in the tenant's seed file -> Migration -> Seed

**Reply by ludralph** [reply]
Hi @baijanathTharu 

it would be really helpful if you could provide the additional information that we had asked about last time. If your problem is resolved by now or somehow doesn't feel relevant any more, you can simply close the Discussion as well.

**Comment by ludralph** [comment, marked as answer]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
